cmd/snap-assert/main.go

```diff
@@ -0,0 +1,129 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
```

Comment:
For the reasons we discussed today, this should probably be inside the "snap" tool itself, as it will talk to snapd for ensuring the content signed is reasonable. "snap sign" sounds nice (even despite that). Then, per above points, let's please simplify its interface to be a bit more tight.

As a strawman, how about `cat input.yaml | snap sign`, where the input looks like
```
type: ...
authority-id: ...
...
revision: ...
body:
```
We don't have to worry about JSON for now.

With that, we can just decode into a map[string]string, take "body" out, and hand onto Database.Sign using the data we can learn directly from it (assertion type, etc).